### DISCUSSION TITLE
where to add weight decay scheduler

### DISCUSSION THREAD

**Discussion by FredZZD** [Lightning-AI/pytorch-lightning#13592] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I want to change weight decay during training, which hook should I overwrite, optimizer_step(), on_train_batch_end() or ... ?

**Comment by akihironitta** [comment, marked as answer]
@FredMushZhao I think either works if you want to schedule it on each step. Alternatively, you could use `on_train_epoch_start` if you want to update it on each epoch.

For reference on all available hooks: https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#hooks

**Reply by FredZZD** [reply]
thanks. :)
